c_src/notif_set_tlh_htable.c

```diff
@@ -0,0 +1,198 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <limits.h>
+#include "notif_set.h"
+#include "optics_htable/htable.h"
+#include "message.h"
+
+struct Notif_set {
+    // a two-level hash table where the value of each
+    // key points to another (level2) hash table 
+    struct htable lev1; // directory watch descriptors
+    size_t len;
+};
+
+struct Notif_set_itr {
+    struct Notif_set *tab;
+    struct htable_bucket *itr1, *itr2;
+};
+
+struct Notif_set *notif_set_new()
+{
+    struct Notif_set *tlh = malloc(sizeof(struct Notif_set));
```

Comment:
I'd recommend using the following: `struct Notif_set *tlh = malloc(sizeof(*tlh));`
Using the size of the variable instead of the size of the type is less error prone (eg. changed the type on the variable declaration but forgot to change the `sizeof` call.

Reply:
Also, what does `tlh` stand for? I can't quite figure it out which probably means that a better name is warranted.

Reply:
Also, I'd recommend allocating like so instead: `calloc(1, sizeof(*tlh));`
This will zero-initialise your entire structure which avoid having uninitialised fields (like you currently do).

Reply:
tlh stands for two-level hash; three-letter acronym :)